app/src/main/java/org/kontalk/ui/view/ComposerBar.java

```diff
@@ -795,6 +809,29 @@ private void hideSoftKeyboard() {
             InputMethodManager.HIDE_IMPLICIT_ONLY);
     }
 
+    private Editable smileyConvert(Editable input){
+        for (String key:emojiMap.keySet()) {
+            input = convert(input, key, emojiMap.get(key));
+        }
+        return input;
+    }
+
+    private Editable convert(Editable text, String in, String out){
+        int position = text.toString().indexOf(in);
+        if(position>-1){
+            return text.replace(position, position + in.length(), out);
+        }
+        else return text;
+    }
```

Comment:
* converting multiple equal emojis is still not supported
* returns not needed

After a quick look at the Android API an `InputFilter` should be used exactly for this case